README.md

```diff
@@ -24,6 +24,23 @@ if __name__ == "__main__":
     grpc_server = UserDefinedFunctionServicer(map_handler=my_handler)
     grpc_server.start()
 ```
+### MapT
```

Comment:
Can we add more docs for:

- What is the `MapT` (also, should we use `MapT` here as the doc title)?
- What is the difference between this and a regular map function, when to use which one.

Reply:
+1. I added some more docs. I can add to this file a link to numaflow transformer doc once we have it.